src/lib.rs

```diff
@@ -3799,46 +3795,96 @@ safety_comment! {
     unsafe_impl_for_power_set!(A, B, C, D, E, F, G, H, I, J, K, L -> M => Immutable for opt_extern_c_fn!(...));
 }
 
-macro_rules! impl_traits_for_atomics {
-    ($($atomics:ident [$inners:ident]),* $(,)?) => {
-        $(
-            impl_for_transparent_wrapper!(TryFromBytes for $atomics [UnsafeCell<$inners>]);
-            impl_for_transparent_wrapper!(FromZeros for $atomics [UnsafeCell<$inners>]);
-            impl_for_transparent_wrapper!(FromBytes for $atomics [UnsafeCell<$inners>]);
-            impl_for_transparent_wrapper!(IntoBytes for $atomics [UnsafeCell<$inners>]);
-        )*
-    };
+/// Implements the zerocopy traits for atomic integer types.
+///
+/// Note that we only implement traits for atomics types if the corresponding
+/// [`#[cfg(target_has_atomic)]`][target_has_atomic] configuration is true.
+///
+/// Note: On targets like `thumbv6m-none-eabi` (which has an [`AtomicU32`]
+/// type but `#[cfg(target_has_atomic = "32")]` is false), we don't implement
+/// the zerocopy traits. We will be able to handle this when Rust stabilizes
+/// [`#[cfg(target_has_atomic_load_store)]`][target_has_atomic_load_store].
```

Comment:
```suggestion
/// TODO(#1009): On targets like `thumbv6m-none-eabi` (which has an [`AtomicU32`]
/// type but `#[cfg(target_has_atomic = "32")]` is false), we don't implement
/// the zerocopy traits. We will be able to handle this when Rust stabilizes
/// [`#[cfg(target_has_atomic_load_store)]`][target_has_atomic_load_store].
```

I've updated #1009 to track this. Feel free to suggest edits to that issue if there's more detail I should include.